### ISSUE TITLE
Using a curl command line to access the mbed-client resource, the connector server replies a "No notification channel found".

### ISSUE THREAD

**Issue by hxingithub** [ARMmbed/mbed-client#192]
when calling:
curl -H 'Authorization: Bearer xxxxxx' https://api.connector.mbed.com/endpoints.  or
curl -v -H 'Authorization: Bearer xxxxxx' https://api.connector.mbed.com/endpoints/endpointName
The connector server replies endpoints information correctly.

When calling:
curl -v -H 'Authorization: Bearer xxxxxx' https://api.connector.mbed.com/endpoints/<endpointname>/Test/0/S,
the connector server replies "No notification channel found".

Is it caused by the asynchronous https response?

Thanks

**Comment by jyrves01** [comment]
Hi, 
This is about the asynchronous stuff. We've noticed that there is a gap in the story, notification channel isn't explained well enough in the developer story, and there are better instructions on the way.

But to help you right now: Connector server may not have endpoint resource values in cache, either those have never been requested, or the ttl set by the resource has expired. In those cases connector server needs to go and ask for the value from endpoint itself. Due to network latencies, low processing power in endpoints and lots of other nasty business it may take time, up to several minutes. We've resolved that with the concept of notification channel. Notification channel is a place you need to set up, which takes in responses to the asynchronous requests. This is somewhat explained in the REST API documentation link on Connector web app Dashboard. Notification channels are unique per the Bearer token (in case you have several webapps, you may want to have the async responses only made from that webapp). In Connector web app we take care for all of this under the hood. Sorry for not telling this clearly enough, but we're improving the instructions.

TLDR; Response is because Async callback channel has not been defined for that Bearer token. See REST API doc section Asynchronous requests.

**Comment by hxingithub** [comment]
Thanks for your reply. It is helpful:)

**Comment by teetak01** [comment]
I presume that the issue was been resolved to satisfaction. Please open a new ticket or comment on this if you have other issues with mbed Client.
